cli/updater/updater.go

```diff
@@ -0,0 +1,138 @@
+// This file is part of arduino-cli.
+//
+// Copyright 2020 ARDUINO SA (http://www.arduino.cc/)
+//
+// This software is released under the GNU General Public License version 3,
+// which covers the main part of arduino-cli.
+// The terms of this license can be found at:
+// https://www.gnu.org/licenses/gpl-3.0.en.html
+//
+// You can be released from the requirements of the above licenses by purchasing
+// a commercial license. Buying such a license is mandatory if you want to
+// modify or otherwise use the software for commercial activities involving the
+// Arduino software without disclosing the source code of your own applications.
+// To purchase a commercial license, send an email to license@arduino.cc.
+
+package updater
+
+import (
+	"os"
+	"strings"
+	"time"
+
+	"github.com/arduino/arduino-cli/cli/feedback"
+	"github.com/arduino/arduino-cli/cli/globals"
+	"github.com/arduino/arduino-cli/configuration"
+	"github.com/arduino/arduino-cli/httpclient"
+	"github.com/arduino/arduino-cli/i18n"
+	"github.com/arduino/arduino-cli/inventory"
+	"github.com/mgutz/ansi"
+	semver "go.bug.st/relaxed-semver"
+)
+
+var tr = i18n.Tr
+
+// CheckForUpdate returns the latest available version if greater than
+// the one running and it makes sense to check for an update, nil in all other cases
+func CheckForUpdate(currentVersion *semver.Version) *semver.Version {
+	if !shouldCheckForUpdate(currentVersion) {
+		return nil
+	}
+
+	return checkForUpdate(currentVersion)
+}
+
+// ForceCheckForUpdate always returns the latest available version if greater than
+// the one running, nil in all other cases
+func ForceCheckForUpdate(currentVersion *semver.Version) *semver.Version {
+	return checkForUpdate(currentVersion)
+}
+
+func checkForUpdate(currentVersion *semver.Version) *semver.Version {
+	defer func() {
+		// Always save the last time we checked for updates at the end
+		inventory.Store.Set("updater.last_check_time", time.Now())
+		inventory.WriteStore()
+	}()
+
+	latestVersion, err := semver.Parse(getLatestRelease())
+	if err != nil {
+		return nil
+	}
+
+	if currentVersion.GreaterThanOrEqual(latestVersion) {
+		// Current version is already good enough
+		return nil
+	}
+
+	return latestVersion
+}
+
+// NotifyNewVersionIsAvailable prints information about the new latestVersion
+func NotifyNewVersionIsAvailable(latestVersion string) {
+	feedback.Errorf("\n\n%s %s → %s\n%s",
+		ansi.Color(tr("A new release of Arduino CLI is available:"), "yellow"),
+		ansi.Color(globals.VersionInfo.VersionString, "cyan"),
+		ansi.Color(latestVersion, "cyan"),
+		ansi.Color("https://arduino.github.io/arduino-cli/latest/installation/#latest-packages", "yellow"))
```

Comment:
`github.com/mattn/go-colorable` is already being used for the verbose output:
https://github.com/arduino/arduino-cli/blob/b8d30dcd169861019f7b5a20916b5521d84867d5/cli/cli.go#L152

Looks like it is proposed that it be used for all the non-verbose output as well:
https://github.com/arduino/arduino-cli/pull/1251/files#diff-ef44b92f1ba3916c39c98fc4a25b67da847cb2a07a5d1253e1978365338888a6

Reply:
I partly fixed this in a separate PR #1418, I'll also need to remove this `ansi` package and use the `color` one that's already used in other parts of the CLI.

Reply:
Fixed, now works fine now: 
![image](https://user-images.githubusercontent.com/3314350/131359000-df253ba0-f23e-4373-ab19-ca9c870a55a1.png)

Reply:
Looks great Silvano! I think it's important to give some care to the Windows users, since my observations indicate it is by far the most common OS for Arduino users in general (I don't know how much that shifts when it comes to CLI users specifically though).